controllers/front/ProductController.php

```diff
@@ -87,6 +87,12 @@ public function init()
             'entity' => 'product',
             'id' => $id_product,
         ));
+        
+        // Get product by EAN-13
+        if ($ean13_product = (int) Tools::getValue('ean13_product')) {
+            $id_product = Db::getInstance()->getValue('SELECT `id_product` FROM `'._DB_PREFIX_.'product` p WHERE p.`ean13` = '. $ean13_product);
+            $id_product = (!$id_product) ? null : $id_product;
```

Comment:
On the technical review, please note the `$id_product` you define here is already needed line 88.

It seems more appropriate to move this a few lines before.